docs: Add common use cases (#4677)
CloudFire-LLC/cloudfire: website/src/app/kb/use-cases/secure-dns/readme.mdx

examples of some popular providers that offer malware-blocking and
privacy-focused DNS resolvers for public use:

| Provider                      | DNS filtering IP(s)                                                  |
| ----------------------------- | -------------------------------------------------------------------- |
| Cloudflare                    | `1.1.1.2`, `1.0.0.2`, `2606:4700:4700::1112`, `2606:4700:4700::1002` |
| Quad9                         | `9.9.9.11`, `149.112.112.11`, `2620:fe::11`, `2620:fe::fe:11`        |
| Comodo Secure DNS             | `8.26.56.26`, `8.20.247.20`                                          |
| dns0.eu                       | `193.110.81.9`, `185.253.5.9`                                        |
| CleanBrowsing Security Filter | `185.228.168.9`, `185.228.169.9`, `2a0d:2a00:1::2`, `2a0d:2a00:2::2` |
| NextDNS                       | Varies. See [NextDNS](https://nextdns.io/) for more information.     |

<Alert color="warning">
  Each provider varies greatly in which domains they block which will affect
  their effectiveness depending on your security requirements. We recommend

>   testing a few providers to see which one works best for you.

Reviewer: ```suggestion
  Each provider blocks different domains. We recommend
  testing a few providers to see which one works best for you.
```

</Alert>

## Step 2: Add the resolver in Firezone

1. [Sign in](https://app.firezone.dev) to your Firezone account and navigate to
   `Settings -> DNS` tab.
1. Add each of the IP addresses of the DNS resolvers you want to use in the
   **Client DNS** section.
1. Click **Save** to apply the changes. The Firezone Client will automatically

>    update its configuration to use the new DNS resolvers.

Reviewer: ```suggestion
1. Click **Save** to apply the changes. The Firezone Clients will automatically
   update their configuration to use the new DNS resolvers.
```

Nit: plural?

<Link
  href="/images/kb/use-cases/secure-dns/step2.png"
  target="_blank"
  rel="nofollow"
>
  <Image
    src="/images/kb/use-cases/secure-dns/step2.png"
    alt="Add the resolver in Firezone"
    width={1200}
    height={1200}
  />
</Link>

<Alert color="info">

>   Firezone currently supports upstream resolvers available over UDP/53 only. See

Reviewer: ```suggestion
  Firezone only supports upstream resolvers available over UDP/53. See
```

  [this GitHub issue](https://github.com/firezone/firezone/issues/4668) if you
  need support for DNS-over-HTTPS (DoH) or DNS-over-TLS (DoT) resolvers.
</Alert>

## Step 3: Verify

You can verify that DNS is being filtered by querying a known malicious domain
from a device with the Firezone Client signed in and checking if you receive an
answer of `0.0.0.0` or `NXDOMAIN`.

For example, the `zeustracker.abuse.ch` domain is associated with the Zeus

> botnet and is commonly blocked by many DNS providers:

Reviewer: That's pretty scary 😨 Should there be an info box "don't actually open that URL in a browser"?

Author: It doesn't actually do anything -- it's just a convenient always-blocked domain that was used to track zeus malware C&C queries
